### PR TITLE
Improve developer documentation around pull request reviews

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Renames old page called ticket testing to refer to reviewing a pull request to be more in line with current practice. Also adds a note to the workflow page on how to checkout a pull request regardless of whether it from the main repository or a fork.

**Report to:** [nobody]. 

**To test:**

Read the updates and see that they make sense.

*There is no associated issue.*

*This does not require release notes* because **it is developer documentation.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/GitWorkflow.rst [hunk 1]
```diff
@@ -94,8 +94,31 @@ When creating a pull request you should:
...
+To check out a particular pull request for functional testing use the following commmands:
+
+.. code-block:: sh
+
+   git fetch <main-remote-name>  +pull/<ID>/merge:pr/<ID>-merged
+   git checkout pr/<ID>-merged
+
+where ``<ID>`` is the pull request number given on GitHub and ``<main-remote-name>`` is the name
+of the remote pointing to the original ``mantid`` repository. If you cloned directly from `mantid <https://github.com/mantidproject/mantid>`_
+then ``main-remote-name=origin`` else if your clone from a fork then it is the name of remove that points
```
Comment (robertapplin) [inline dev-docs/source/GitWorkflow.rst]: you instead of your?
remote instead of remove?

File: dev-docs/source/GitWorkflow.rst [hunk 2]
```diff
@@ -94,8 +94,31 @@ When creating a pull request you should:
...
+back to the original repository.
+
+Note that these commands will checkout about temporary branch that has the development branch merged with master and note just
```
Comment (robertapplin) [inline dev-docs/source/GitWorkflow.rst]: not instead of note (at the end of the line)

**Review by robertapplin — CHANGES_REQUESTED** [review]
The changes look good, there are just a few typos I believe

File: dev-docs/source/GitWorkflow.rst (hunk 1 above)
Comment (martyngigg) [reply dev-docs/source/GitWorkflow.rst]: Nice catch!

File: dev-docs/source/GitWorkflow.rst (hunk 2 above)
Comment (martyngigg) [reply dev-docs/source/GitWorkflow.rst]: Nice catch!

**Comment by martyngigg** [comment]
Thanks for spotting those mistakes, Rob. They should be fixed now.

**Review by robertapplin — APPROVED** [review]

**Pull request merged**